### ISSUE TITLE
Integer parameter value changed to floating type

### ISSUE THREAD

**Issue by 0liu** [SimonBlanke/Gradient-Free-Optimizers#15]
**Describe the bug**
Integer parameter values defined in search space are changed to floating type during optimization.

**Code to reproduce the behavior**
```python
search_space = dict(my_para = range(100, 200, 10))
opt = RandomSearchOptimizer(search_space)
opt.search(obj_func, n_iter=100)

# inside obj_func
def obj_func(my_para):
    for i in range(my_para):
        do_something()
    return score
```

**Error message from command line**
```TypeError: 'numpy.float64' object cannot be interpreted as an integer```

**System information:**
 - Ubuntu 21.04
 - Python version: 3.9.4
 - Gradient-Free-Optimizers version: 0.3.2

**Additional context**

**Comment by SimonBlanke** [comment]
This issue is reopened because if a related problem in [Hyperactive](https://github.com/SimonBlanke/Hyperactive/issues/34).

The error can be replicated with this example:
```python
from gradient_free_optimizers import RandomSearchOptimizer

search_space = dict(my_para=range(100, 200, 10), my_para_2=[0.1, 0.2, 0.3])

def obj_func(para):
    score = 1
    my_para = para["my_para"]
    print("my_para", my_para)
    for i in range(my_para):
        pass
    return score

opt = RandomSearchOptimizer(search_space)
opt.search(obj_func, n_iter=100)
```

**Comment by SimonBlanke** [comment]
@0liu,

the fix of the problem in [issue 34 of Hyperactive](https://github.com/SimonBlanke/Hyperactive/issues/34) seems very similar to this problem here, because the GFO [Converter-class](https://github.com/SimonBlanke/Gradient-Free-Optimizers/blob/master/gradient_free_optimizers/converter.py) contains very similar methods (position2value, ...). Maybe you could fix the problem in GFO in a pull request. If you do so, make sure to test it in a simple script.

**Comment by 0liu** [comment]
Yes I can make similar PR for GFO. Allow me a few days delay, maybe next week. Thanks.

**Comment by SimonBlanke** [comment]
Thanks @0liu for the pull request SimonBlanke/Gradient-Free-Optimizers#17. I wrote some additional fixes in 13821b1a2c4036ebab62dbbf8db23423d4db7982 and moved your tests in d0ef41b5f2327c50e055e8f59c9ba0b749b2e9f6. I also tested these changes against the current Hyperactive API and it all seems to work fine (but I did not test SimonBlanke/Hyperactive#34, yet).
